client/state/login/actions.js

```diff
@@ -195,11 +195,15 @@ export const loginSocialUser = ( service, token ) => dispatch => {
 		.send( {
 			service,
 			token,
+			redirect_to: redirectTo,
 			client_id: config( 'wpcom_signup_id' ),
 			client_secret: config( 'wpcom_signup_key' ),
 		} )
-		.then( () => {
-			dispatch( { type: SOCIAL_LOGIN_REQUEST_SUCCESS } );
+		.then( ( response ) => {
+			dispatch( {
+				type: SOCIAL_LOGIN_REQUEST_SUCCESS,
+				redirectTo: get( response, 'body.data.redirect_to' ),
```

Comment:
It's using `window.location.origin` when redirecting if we have empty value stored in Redux. It's better because it works with all different domains we use.